src/storage/DataAccessorBasedStore.ts

```diff
@@ -287,13 +293,22 @@ export class DataAccessorBasedStore implements ResourceStore {
     // Solid, §5.3: "Servers MUST create intermediate containers and include corresponding containment triples
     // in container representations derived from the URI path component of PUT and PATCH requests."
     // https://solid.github.io/specification/protocol#writing-resources
-    if (createContainers && !this.identifierStrategy.isRootContainer(identifier)) {
-      await this.createRecursiveContainers(this.identifierStrategy.getParentContainer(identifier));
+    const modified = [];
+    if (!this.identifierStrategy.isRootContainer(identifier)) {
+      const container = this.identifierStrategy.getParentContainer(identifier);
```

Comment:
It is possible to determine here if the parent container was modified or not without doing any requests (we check for existense in setRepresentation), so it would be possible to return an exact list of identifiers that actually got modified. The counter argument for this is I think that we don't want to expose if a resource got newly created or not? But is this the place to make that choice? I think it would be more accurate to return accurate information here, and then let the MonitoringStore decide what to expose of that.

Reply:
Indeed; wether or not we expose that over HTTP, we can still choose. But it's useful if the internal interfaces know this.

> It is possible to determine here if the parent container was modified or not without doing any requests (we check for existense in setRepresentation)

How would you do this, pass an extra parameter to `writeData` then?

Alternative: have this as a return value from `writeContainer` and `writeDocument`? Or too expensive there?

---

Note that we don't _have_ to solve this because of how I've defined the semantics in `ResourceStore`. Also begs the question if a container is considered changed only if containment triples change (as opposed to mtime, size, etc.).

Reply:
I would not change anything to the DataAccessors in this PR so would not do it there. Could either be a boolean in `writeData` or do it specifically in the `setRepresentation`/`addResource` functions. Something similar could also be done in the Patch handler since there we also check if the target resource already exists or not.

> Also begs the question if a container is considered changed only if containment triples change (as opposed to mtime, size, etc.).

Yea that's the discussion of course. Would be nice if this was strictly defined though.

Reply:
> I would not change anything to the DataAccessors

Agreed.